scenario/runtime.py

```diff
@@ -298,38 +295,48 @@ def _virtual_charm_root(self) -> typing.ContextManager[Path]:
         config_yaml = virtual_charm_root / "config.yaml"
         actions_yaml = virtual_charm_root / "actions.yaml"
 
-        metadata_files_present = any(
-            file.exists() for file in (metadata_yaml, config_yaml, actions_yaml)
-        )
+        metadata_files_present: Dict[Path, Union[str, False]] = {
+            file: file.read_text() if file.exists() else False
+            for file in (metadata_yaml, config_yaml, actions_yaml)
+        }
+
+        any_metadata_files_present_in_vroot = any(metadata_files_present.values())
```

Comment:
What if the file exists but is empty string (falsey)? Probably better to be more explicit:

```suggestion
        any_metadata_files_present_in_vroot = any(v is not None for v in metadata_files_present.values())
```